src/extension-runners/firefox-desktop.js

```diff
@@ -217,22 +217,21 @@ export class FirefoxDesktopExtensionRunner {
       startUrl,
       firefoxApp,
       firefoxClient,
+      args = [],
     } = this.params;
 
-    const binaryArgs = [];
```

Comment:
In this version of the PR the arguments are still prepended instead of appended. Restore the `binaryArgs` variable and use `binaryArgs.push(...args);` before passing it to `firefoxApp.run`.

Reply:
Oh, looks like I misunderstood you. Does it look fine now - https://github.com/mozilla/web-ext/pull/1563/commits/28cb262c13bdf627b5f3076eca9fc705149c5f37?